routing/router.go

```diff
@@ -111,8 +114,8 @@ type Config struct {
 	// forward a fully encoded payment to the first hop in the route
 	// denoted by its public key. A non-nil error is to be returned if the
 	// payment was unsuccessful.
-	SendToSwitch func(firstHop *btcec.PublicKey,
-		htlcAdd *lnwire.UpdateAddHTLC) ([32]byte, error)
+	SendToSwitch func(firstHop *btcec.PublicKey, htlcAdd *lnwire.UpdateAddHTLC,
```

Comment:
Layer violation. The router shouldn't have to know about the interfaces of the switch, or that encrypted errors are even used at all. 

Instead something like the following should be added as a parameter:
```go
type Circuit struct {
    SessionKey *btcec.PrivateKey

    Path []*btcec.PublicKey
}
```

Such a struct should then be added to the `sphinx` package. In the server, when creating the `ChannelRouter`, the closure there should then morph the `sphinx.Circuit` instance into the `htlcswitch.Deobfuscator`. A little indirection goes a long way.